Building FMUs/fmusim on aarch64-darwin

@t-sommer  I was able to build 'fmusim' and reference FMUs on ARM M2 Mac. When running build.py I had to fix the following:
`Reference-FMUs/src/fmi3Functions.c:241:27: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
const char* fmi3GetVersion() {`                                       
by changing it to `const char* fmi3GetVersion(void)`.
There were also a bunch of warnings. Let me know if you want the list.

--- a/src/fmi3Functions.c
+++ b/src/fmi3Functions.c
@@ -238,7 +238,7 @@ static bool allowedState(ModelInstance *instance, int statesExpected, char *name
  Common Functions
  ****************************************************/
 
-const char* fmi3GetVersion() {
+const char* fmi3GetVersion(void) {
     return fmi3Version;
 }
 

